src/main/java/org/apache/brooklyn/cloudfoundry/entity/VanillaCloudfoundryApplication.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.brooklyn.cloudfoundry.entity;
+
+
+import org.apache.brooklyn.api.entity.Entity;
+import org.apache.brooklyn.api.entity.ImplementedBy;
+import org.apache.brooklyn.api.sensor.AttributeSensor;
+import org.apache.brooklyn.cloudfoundry.location.CloudFoundryPaasLocation;
+import org.apache.brooklyn.config.ConfigKey;
+import org.apache.brooklyn.core.config.ConfigKeys;
+import org.apache.brooklyn.core.config.MapConfigKey;
+import org.apache.brooklyn.core.entity.Attributes;
+import org.apache.brooklyn.core.entity.BrooklynConfigKeys;
+import org.apache.brooklyn.core.entity.lifecycle.Lifecycle;
+import org.apache.brooklyn.core.entity.trait.Startable;
+import org.apache.brooklyn.core.sensor.Sensors;
+import org.apache.brooklyn.util.collections.MutableMap;
+import org.apache.brooklyn.util.core.flags.SetFromFlag;
+import org.apache.brooklyn.util.time.Duration;
+
+@ImplementedBy(VanillaCloudfoundryApplicationImpl.class)
+public interface VanillaCloudfoundryApplication extends Entity, Startable {
+
+    @SetFromFlag("name")
+    ConfigKey<String> APPLICATION_NAME = ConfigKeys.newStringConfigKey(
+            "cloudFoundry.application.name", "Name of the application");
+
+    @SetFromFlag("path")
+    ConfigKey<String> ARTIFACT_PATH = ConfigKeys.newStringConfigKey(
+            "cloudFoundry.application.artifact", "URI of the application");
+
+
+    @SetFromFlag("buildpack")
+    ConfigKey<String> BUILDPACK = ConfigKeys.newStringConfigKey(
+            "cloudFoundry.application.buildpack", "Buildpack to deploy an application");
+
+    @SetFromFlag("env")
+    MapConfigKey<String> ENVS = new MapConfigKey<String>(String.class, "cloudfoundry.application.env",
+            "Enviroment variables for the application", MutableMap.<String, String>of());
+
+    @SetFromFlag("startTimeout")
+    ConfigKey<Duration> START_TIMEOUT = BrooklynConfigKeys.START_TIMEOUT;
+
+    AttributeSensor<String> ROOT_URL =
+            Sensors.newStringSensor("webapp.url", "URL of the application");
+
+    AttributeSensor<CloudFoundryPaasLocation> CLOUDFOUNDRY_LOCATION = Sensors.newSensor(
```

Comment:
I added a sensor to expose the used `CloudFoundryPaasLocation` because I found that `SoftwareProcess` contains an sensor to show the machine where it is running, [take a look here](https://github.com/apache/brooklyn-server/blob/master/software/base/src/main/java/org/apache/brooklyn/entity/software/base/SoftwareProcess.java#L326-L327). However, not I am thinking that `SoftwareProcess` uses this sensor because the entity could receives a `ProvisioningLocation` as as location which is used to provision a machine. Then in this case the machine is obtained in runtime ( and the sensor is used to show the generated location where the entity is running. However, `CloudFoundryPaasLocation` is a static location, and any new location is obtained. So, it would not make sense show the used `CloudFoundryLocation` as a sensor.